src/com/qamp/HarisJasarevic/homeworks/homework5/task1/BankAccounts.java

```diff
@@ -0,0 +1,21 @@
+package com.qamp.HarisJasarevic.homeworks.homework5.task1;
+
+public abstract class BankAccounts {
+
+    protected AccountOwner accountOwner;
+    protected String serialNumber;
+    protected int passCode;
+    protected double balance;
+
+    public BankAccounts(final AccountOwner accountOwner, final String serialNumber, final int passCode) {
+        this.accountOwner = accountOwner;
+        this.serialNumber = serialNumber;
+        this.passCode = passCode;
+        this.balance = 0;
+    }
+
+    public abstract double moneyWithdraw ( final double withdrawAmount) throws IllegalStateException;
+    public abstract double moneyDeposit (final  double depositAmount) throws IllegalStateException;
+    public abstract void reset ();
+    public abstract String showAccountState ();
```

Comment:
As per my other comments, consider the possibility to make these concrete methods.

I believe you should be able to reuse their behaviour. 
Also since CurrentAccount reset() does nothing it seems like a reasonable default behaviour for any future account that might come up, but this is hard to tell since the example really has only 2 now.